## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 502801`
Commit: [`cb544cdfbf7000bdf4a646d254a31f7ed8d22857`](https://github.com/NixOS/nixpkgs/commit/cb544cdfbf7000bdf4a646d254a31f7ed8d22857) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/cb544cdfbf7000bdf4a646d254a31f7ed8d22857..pull/502801/head))
Merge: [`4504136ff6ffb0d3802680b81a729ef195785650`](https://github.com/NixOS/nixpkgs/commit/4504136ff6ffb0d3802680b81a729ef195785650)

Logs: https://github.com/JamieMagee/nixpkgs-review-gha/actions/runs/23464157424


---
### `x86_64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>tests.home-assistant-component-tests.tuya</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python313Packages.tuya-device-handlers</li>
    <li>python313Packages.tuya-device-handlers.dist</li>
    <li>python314Packages.tuya-device-handlers</li>
    <li>python314Packages.tuya-device-handlers.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>tests.home-assistant-component-tests.tuya</summary>
<pre>=============================== warnings summary ===============================
homeassistant/runner.py:176
  /build/source/homeassistant/runner.py:176: DeprecationWarning: &#x27;asyncio.DefaultEventLoopPolicy&#x27; is deprecated and slated for removal in Python 3.16
    class HassEventLoopPolicy(asyncio.DefaultEventLoopPolicy):  # type: ignore[name-defined,misc]

tests/conftest.py:155
  /build/source/tests/conftest.py:155: DeprecationWarning: &#x27;asyncio.set_event_loop_policy&#x27; is deprecated and slated for removal in Python 3.16
    asyncio.set_event_loop_policy(runner.HassEventLoopPolicy(False))

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
--------------------------- snapshot report summary ----------------------------
4 snapshots failed. 2224 snapshots passed. 177 snapshots unused.

Re-run pytest with --snapshot-update to delete unused snapshots.
=========================== short test summary info ============================
FAILED tests/components/tuya/test_diagnostics.py::test_entry_diagnostics[rqbj_4iqe2hsfyd86kwwc] - AssertionError: assert [+ received] == [- snapshot]
FAILED tests/components/tuya/test_diagnostics.py::test_device_diagnostics[mal_gyitctrjj1kefxp2] - AssertionError: assert [+ received] == [- snapshot]
FAILED tests/components/tuya/test_diagnostics.py::test_device_diagnostics[tdq_9htyiowaf5rtdhrv] - AssertionError: assert [+ received] == [- snapshot]
FAILED tests/components/tuya/test_select.py::test_platform_setup_and_discovery - AssertionError: assert [+ received] == [- snapshot]
================== 4 failed, 141 passed, 2 warnings in 30.37s ==================</pre>
</details>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 1 package failed to build:</summary>
  <ul>
    <li>tests.home-assistant-component-tests.tuya</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python313Packages.tuya-device-handlers</li>
    <li>python313Packages.tuya-device-handlers.dist</li>
    <li>python314Packages.tuya-device-handlers</li>
    <li>python314Packages.tuya-device-handlers.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>tests.home-assistant-component-tests.tuya</summary>
<pre>2026-03-23 23:00:43.709 INFO     MainThread homeassistant.helpers.entity_registry:entity_registry.py:1260 Registered new select.tuya entity: select.cbe_pro_2_inverter_work_mode
=============================== warnings summary ===============================
homeassistant/runner.py:176
  /build/source/homeassistant/runner.py:176: DeprecationWarning: &#x27;asyncio.DefaultEventLoopPolicy&#x27; is deprecated and slated for removal in Python 3.16
    class HassEventLoopPolicy(asyncio.DefaultEventLoopPolicy):  # type: ignore[name-defined,misc]

tests/conftest.py:155
  /build/source/tests/conftest.py:155: DeprecationWarning: &#x27;asyncio.set_event_loop_policy&#x27; is deprecated and slated for removal in Python 3.16
    asyncio.set_event_loop_policy(runner.HassEventLoopPolicy(False))

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
--------------------------- snapshot report summary ----------------------------
3 snapshots failed. 2225 snapshots passed. 177 snapshots unused.

Re-run pytest with --snapshot-update to delete unused snapshots.
=========================== short test summary info ============================
FAILED tests/components/tuya/test_diagnostics.py::test_entry_diagnostics[rqbj_4iqe2hsfyd86kwwc] - AssertionError: assert [+ received] == [- snapshot]
FAILED tests/components/tuya/test_diagnostics.py::test_device_diagnostics[tdq_9htyiowaf5rtdhrv] - AssertionError: assert [+ received] == [- snapshot]
FAILED tests/components/tuya/test_select.py::test_platform_setup_and_discovery - AssertionError: assert [+ received] == [- snapshot]
================== 3 failed, 142 passed, 2 warnings in 28.47s ==================</pre>
</details>
</details>

---
### `x86_64-darwin` (sandbox = relaxed)
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python313Packages.tuya-device-handlers</li>
    <li>python313Packages.tuya-device-handlers.dist</li>
    <li>python314Packages.tuya-device-handlers</li>
    <li>python314Packages.tuya-device-handlers.dist</li>
  </ul>
</details>

---
### `aarch64-darwin` (sandbox = relaxed)
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python313Packages.tuya-device-handlers</li>
    <li>python313Packages.tuya-device-handlers.dist</li>
    <li>python314Packages.tuya-device-handlers</li>
    <li>python314Packages.tuya-device-handlers.dist</li>
  </ul>
</details>

--- a/pkgs/development/python-modules/tuya-device-handlers/default.nix
+++ b/pkgs/development/python-modules/tuya-device-handlers/default.nix
@@ -1,0 +1,41 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+  poetry-core,
+  tuya-device-sharing-sdk,
+  pytestCheckHook,
+  syrupy,
+}:
+
+buildPythonPackage (finalAttrs: {
+  pname = "tuya-device-handlers";
+  version = "0.0.10";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "home-assistant-libs";
+    repo = "tuya-device-handlers";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-W5aSEt8xXxQUcs6+AVVcgXxjm3WppzfCaww8YX+sej0=";
+  };
+
+  build-system = [ poetry-core ];
+
+  dependencies = [ tuya-device-sharing-sdk ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    syrupy
+  ];
+
+  pythonImportsCheck = [ "tuya_device_handlers" ];
+
+  meta = {
+    description = "Tuya quirks library";
+    homepage = "https://github.com/home-assistant-libs/tuya-device-handlers";
+    changelog = "https://github.com/home-assistant-libs/tuya-device-handlers/releases/tag/v${finalAttrs.version}";
+    license = lib.licenses.mit;
+    maintainers = [ lib.maintainers.jamiemagee ];
+  };
+})

--- a/pkgs/development/python-modules/tuya-device-sharing-sdk/default.nix
+++ b/pkgs/development/python-modules/tuya-device-sharing-sdk/default.nix
@@ -10,14 +10,14 @@
 
 buildPythonPackage (finalAttrs: {
   pname = "tuya-device-sharing-sdk";
-  version = "0.2.9";
+  version = "0.2.8";
   pyproject = true;
 
   src = fetchFromGitHub {
     owner = "tuya";
     repo = "tuya-device-sharing-sdk";
     tag = finalAttrs.version;
-    hash = "sha256-kNWg+AXISThwK14ByObUr+/4GMntrZgtEEMNpw/HjLw=";
+    hash = "sha256-uD2Lzs08i/01iEksXpYRGB2lU7I15sQrJLfJZ93+oeY=";
   };
 
   build-system = [ setuptools ];

--- a/pkgs/servers/home-assistant/component-packages.nix
+++ b/pkgs/servers/home-assistant/component-packages.nix
@@ -6527,8 +6527,9 @@
     "tuya" =
       ps: with ps; [
         ha-ffmpeg
+        tuya-device-handlers
         tuya-device-sharing-sdk
-      ]; # missing inputs: tuya-device-handlers
+      ];
     "twentemilieu" =
       ps: with ps; [
         twentemilieu
@@ -8196,6 +8197,7 @@
     "trend"
     "triggercmd"
     "tts"
+    "tuya"
     "twentemilieu"
     "twilio"
     "twinkly"

--- a/pkgs/servers/home-assistant/tests.nix
+++ b/pkgs/servers/home-assistant/tests.nix
@@ -204,6 +204,11 @@ let
       # TraneConfigFlow doesn't support step reauth
       "tests/components/trane/test_init.py::test_setup_auth_error"
     ];
+    tuya = [
+      # entity ordering in diagnostics is non-deterministic; fixed upstream in
+      # https://github.com/home-assistant/core/pull/164819 (landing in HA 2026.4)
+      "tests/components/tuya/test_diagnostics.py"
+    ];
     youtube = [
       # outdated snapshot
       "tests/components/youtube/test_sensor.py::test_sensor"
@@ -236,6 +241,12 @@ let
     shell_command = [
       # tries to retrieve file from github
       "test_non_text_stdout_capture"
+    ];
+    tuya = [
+      # snapshot mismatches: PyPI sdist translations differ from strings.json
+      # ("Power-on behavior" vs "Power on behavior"); expected to resolve in HA 2026.4
+      "test_device_diagnostics[tdq_9htyiowaf5rtdhrv]"
+      "test_platform_setup_and_discovery"
     ];
     zeroconf = [
       # multicast socket bind, not possible in the sandbox

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -19758,6 +19758,8 @@ self: super: with self; {
 
   turrishw = callPackage ../development/python-modules/turrishw { };
 
+  tuya-device-handlers = callPackage ../development/python-modules/tuya-device-handlers { };
+
   tuya-device-sharing-sdk = callPackage ../development/python-modules/tuya-device-sharing-sdk { };
 
   tuya-iot-py-sdk = callPackage ../development/python-modules/tuya-iot-py-sdk { };

